doc/stream-telemetry/stream-telemetry-hld.md

```diff
@@ -0,0 +1,930 @@
+# Stream telemetry high level design #
+- [Stream telemetry high level design](#stream-telemetry-high-level-design)
+  - [Table of Content](#table-of-content)
+    - [Revision](#revision)
+    - [Scope](#scope)
+    - [Definitions/Abbreviations](#definitionsabbreviations)
+    - [Overview](#overview)
+    - [Requirements](#requirements)
+    - [Architecture Design](#architecture-design)
+    - [High-Level Design](#high-level-design)
+      - [Modules](#modules)
+        - [Netlink Module](#netlink-module)
+      - [Data format](#data-format)
+        - [IPFIX header](#ipfix-header)
+        - [IPFIX template](#ipfix-template)
+        - [IPFIX data](#ipfix-data)
+        - [Bandwidth Estimation](#bandwidth-estimation)
+      - [Config DB](#config-db)
+        - [STREAM\_TELEMETRY\_PROFILE](#stream_telemetry_profile)
+        - [STREAM\_TELEMETRY\_GROUP](#stream_telemetry_group)
+      - [StateDb](#statedb)
+        - [STREAM\_TELEMETRY\_SESSION](#stream_telemetry_session)
+      - [Work Flow](#work-flow)
+    - [SAI API](#sai-api)
+      - [Create HOSTIF object](#create-hostif-object)
+      - [Creating TAM transport object](#creating-tam-transport-object)
+      - [Creating TAM collector object](#creating-tam-collector-object)
+      - [Creating TAM report object](#creating-tam-report-object)
+      - [Creating TAM telemetry type object](#creating-tam-telemetry-type-object)
+      - [Creating TAM telemetry object](#creating-tam-telemetry-object)
+      - [Create TAM counter subscription objects](#create-tam-counter-subscription-objects)
+      - [Create TAM object](#create-tam-object)
+      - [Query IPFIX template](#query-ipfix-template)
+      - [Enable/Disable telemetry stream](#enabledisable-telemetry-stream)
+    - [Configuration and management](#configuration-and-management)
+      - [Manifest (if the feature is an Application Extension)](#manifest-if-the-feature-is-an-application-extension)
+      - [CLI/YANG model Enhancements](#cliyang-model-enhancements)
+      - [Config DB Enhancements](#config-db-enhancements)
+    - [Warmboot and Fastboot Design Impact](#warmboot-and-fastboot-design-impact)
+    - [Memory Consumption](#memory-consumption)
+    - [Restrictions/Limitations](#restrictionslimitations)
+    - [Testing Requirements/Design](#testing-requirementsdesign)
+      - [Unit Test cases](#unit-test-cases)
+      - [System Test cases](#system-test-cases)
+    - [Open/Action items - if any](#openaction-items---if-any)
+
+
+## Table of Content 
+
+### Revision  
+
+### Scope  
+
+This section describes the scope of this high-level design document in SONiC.
+
+### Definitions/Abbreviations 
+
+This section covers the abbreviation if any, used in this high-level design document and its definitions.
```

Comment:
are these sections empty? if so, it will be better to put some other word instead of these "auther-guidance-words". this is very confusing, especially all section headers doesn't have any numbered prefix, it is very easy to lose when reading the doc.